### PR TITLE
Show Error when Invalid names Filter (Repo and Catalog)

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- When filtering Repo and Catalogs if there is an invalid Name, we should show an error. 
- When filtering on Catalogs name filter, should have at least 1 value. This change is to be consistent with the other datasources.

# ISSUE TYPE
- Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME
- Repository Datasource
- Catalog Datasource

#### Test Output
```
$ MOCKEY_CHECK_GCFLAGS=false go test -v -timeout 1h -run TestDataSource_FirmwareCatalogRead
=== RUN   TestDataSource_FirmwareCatalogRead
--- PASS: TestDataSource_FirmwareCatalogRead (16.64s)
PASS
ok      terraform-provider-ome/ome      20.656s

$ MOCKEY_CHECK_GCFLAGS=false go test -v -timeout 1h -run TestDataSource_Fbc_Repository_Read
=== RUN   TestDataSource_Fbc_Repository_Read
--- PASS: TestDataSource_Fbc_Repository_Read (17.37s)
PASS
ok      terraform-provider-ome/ome      20.010s
```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by rounak-adhikary — DISMISSED** [review]

File: examples/data-sources/ome_firmware_catalog/data-source.tf [hunk 1]
```diff
@@ -14,8 +14,8 @@ limitations under the License.
 # Get details of all the firmware catalogs 
 data "ome_firmware_catalog" "data-catalog" {
   # Can filter based on the catalog name
-  # If empty then all catalogs are returned
-  # If invalid name is set that will be ignored
+  # If at least one name is invalid names will return an error
```
Comment (shekhar-j) [inline examples/data-sources/ome_firmware_catalog/data-source.tf:17]: @doriac11 can you rephrase this?

**Review by shekhar-j — APPROVED** [review]

File: examples/data-sources/ome_firmware_catalog/data-source.tf (hunk 1 above)
Comment (doriac11) [reply examples/data-sources/ome_firmware_catalog/data-source.tf:17]: Yes!

**Review by nikitajoshi1 — DISMISSED** [review]

**Review by shekhar-j — APPROVED** [review]

**Review by nikitajoshi1 — APPROVED** [review]
